src/Symfony/Component/DependencyInjection/Loader/FileLoader.php

```diff
@@ -85,21 +85,22 @@ protected function setDefinition($id, Definition $definition)
         }
     }
 
-    private function findClasses($namespace, $pattern, $excludePattern)
+    private function findClasses($namespace, $pattern, $excludePatterns)
     {
         $parameterBag = $this->container->getParameterBag();
 
         $excludePaths = array();
         $excludePrefix = null;
-        if ($excludePattern) {
-            $excludePattern = $parameterBag->unescapeValue($parameterBag->resolveValue($excludePattern));
-            foreach ($this->glob($excludePattern, true, $resource) as $path => $info) {
-                if (null === $excludePrefix) {
-                    $excludePrefix = $resource->getPrefix();
+        if ($excludePatterns) {
+            $excludePatterns = (array) $parameterBag->unescapeValue($parameterBag->resolveValue($excludePatterns));
```

Comment:
I'd rather do the cast in `registerClasses` and add the `array $excludePatterns` typehint in this method

Reply:
It's done